### ISSUE TITLE
Unable to call `makeExecutableSchema` due to apollo-server-core dependencies

### ISSUE THREAD

**Issue by jneate** [apollographql/apollo-server#5571]
I suspect that #5553 will fix this issue but raising it anyway as it's not yet released and I could be wrong.

Apollo server core has a dependency on @graphql-tools:

- mock
- schema
- utils

Schema and Utils are on v7 but mock is on v8.

The dependencies of mock v8 are schema and utils v8 meaning that in the package-lock.json file I have both v7 and v8 versions of `@graphql-tools/schema` and `@graphql-tools/utils`

I'm using NodeJS v12 so I can't override/force resolve the dependency used and I'm hitting [this issue](https://github.com/ardatan/graphql-tools/issues/3250)

If there is a way round this before the next release of apollo-server-core then please let me know, or if I've missed something obvious.

Using apollo-server-core v3.1.1 for reference.

**Comment by glasser** [comment]
> The dependencies of mock v8 are schema and utils v8

Well, not exactly... mock v8 came out half a year ago and schema v8 came out this week.  I think the change specifically was in `@graphql-tools/mock@8.1.4`. So you could do some package-lock magic to stay on 8.1.3.

That said I think you're right that #5553 fixes this issue. I'll put out v3.1.2 right now.

**Comment by jneate** [comment]
Thanks @glasser appreciate the quick help/response

**Comment by AhmedBHameed** [comment]
I'm not sure what can solve this issue!! I changed `package-lock` versions for `@graphql-tools/mock` as suggested but still not working.

@jneate Have you solved the issue ?

**Comment by jneate** [comment]
> I'm not sure what can solve this issue!! I changed `package-lock` versions for `@graphql-tools/mock` as suggested but still not working.
> 
> @jneate Have you solved the issue ?

@AhmedBHameed yeah it's working fine on my side.

Upgraded the `apollo-server-core` dependency to version `3.1.2`
Added dependency for `@graphql-tools/schema` at version `8.0.1`

There was also an issue with graphql-tools v8.0.1 using TypeScript and Nodev12 see [this discussion](https://github.com/ardatan/graphql-tools/discussions/3295)

TL;DR:- TypeScript needed to be `>=4.3.0` and I had to add `es2021.promise` to the tsconfig lib section, but this should be fixed in a newer version of `@graphql-tools/schema` I've just not upgraded beyond `8.0.1` yet.

Hopefully this helps

**Comment by AhmedBHameed** [comment]
Thank you for the rapid reply. Appreciate your support.

I did the following

```json
"dependencies": {
    "@graphql-tools/schema": "8.0.1",
    "@hapi/joi": "^17.1.1",
    "apollo-server-core": "^3.1.2",  <<== not sure if i need this because i'm using apollo-server-express
    "apollo-server-express": "^3.1.2",
    "bluebird": "^3.7.2",
    "cookie-parser": "^1.4.5",
    "cors": "^2.8.5",
    "dataloader": "^2.0.0",
    "dotenv": "^10.0.0",
    "express": "^4.17.1",
    "graphql": "^15.5.1",
    "helmet": "^4.6.0",
    "lodash": "^4.17.21",
    "reflect-metadata": "^0.1.13",
    "winston": "^3.3.3",
    "winston-transport": "^4.4.0"
  },
  "devDependencies": {
	[...]
    "@types/node": "^16.4.13",
	[...]
  },
```

The import is falling in this line:

```ts
import {ApolloServer, makeExecutableSchema} from 'apollo-server-express';  <<== red line under makeExecutableSchema
```

my tsconfig.json
```json
{
  "compilerOptions": {
    "module": "commonjs",
    "resolveJsonModule": true,
    "esModuleInterop": true,
    "target": "es6",
    "noImplicitAny": false,
    "moduleResolution": "node",
    "removeComments": true,
    "sourceMap": true,
    "outDir": "dist",
    "noImplicitThis": false,
    "baseUrl": ".",
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "lib": ["es6", "dom"],
    "types": ["reflect-metadata", "@types/jest"],
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules", "**/__tests__/*"]
}
```

And got confused the issue still there!!!
Any suggestions please?

**Comment by jneate** [comment]
@AhmedBHameed I've popped out for lunch so I can't check but I'm fairly sure I'm importing makeExecutableSchema from the graphql-tools/schema package

I think as part of apollo-server V3 they don't export the method anymore

**Comment by AhmedBHameed** [comment]
@jneate You are correct. That solves the issue.
Thanks a lot and have a nice lunch time.
